.tekton/tasks/ec-checks.yaml

```diff
@@ -49,5 +49,25 @@ spec:
       policy='./policies/build-tasks.yaml'
 
       ec validate input --policy "${policy}" --output yaml --strict=true ${args[*]}
+  - name: validate-step-actions
+    workingDir: "$(workspaces.source.path)/source"
+    image: quay.io/enterprise-contract/ec-cli:snapshot@sha256:46fae4d356d678383a926de8a68f79177d7e685d5497675acf41c9d3425aaacc
+    script: |
+      #!/bin/bash
+      set -euo pipefail
+
+      # Generate list of file parameters, e.g. --file foo.yaml --file bar.yaml
+      args=()
+      while IFS= read -r -d '' f; do
+          found="$(yq eval '.kind == "StepAction"' "${f}")"
+          if [[ ${found} == "true" ]]; then
+              args+=( '--file' "${f}" )
+          fi
+      done < <(find stepactions -name '*.yaml' -print0)
```

Comment:
Could do this:
```suggestion
      shopt -s globstar
      readarray -t step_actions < <(yq -r 'select(.kind == "StepAction") | filename' stepactions/**/*.yaml)
      readarray args < <(printf -- " --file %s" "${step_actions[@]}")
```

Reply:
Didn't know about yq's filename! Very useful. Used a variation of this.